app/models/pipeline_run_stage.rb

```diff
@@ -201,7 +201,7 @@ def host_filtering_command
     file_type = sample.input_files.first.file_type
     batch_command_env_variables = "INPUT_BUCKET=#{sample.sample_input_s3_path} OUTPUT_BUCKET=#{sample.sample_output_s3_path} " \
       "FILE_TYPE=#{file_type} DB_SAMPLE_ID=#{sample.id} " \
-      "COMMIT_SHA_FILE=#{COMMIT_SHA_FILE_ON_WORKER} "
+      "COMMIT_SHA_FILE=#{COMMIT_SHA_FILE_ON_WORKER} PIPELINE_VERSION=\\`idseq_pipeline --version\\` "
```

Comment:
This is a little bit too clever :)   What you actually need to capture is the git hash at the top of the pipeline branch being submitted, so that each pipeline stage will checkout that exact hash.   The pipeline version doesn't need to be passed, should be available as a variable to the actual pipeline code.

Reply:
So line 23 in this file, `git checkout #{pipeline_run.pipeline_branch};` needs to change so that it checks out a specific git hash.   This hash can be determined by rev-parse as on line 24 at the time of creating the batch command.

Reply:
The command you need to run in ruby for this is 
```
pipeline_branch ||= "master"
`git ls-remote https://github.com/chanzuckerberg/idseq-pipeline.git | grep refs/heads/#{pipeline_branch}`.split[0]
```

Reply:
i need the pipeline_version to determine the output directory. it's also only give the major version.

Reply:
Right, I understand the pipeline code will emit a pipeline_version.txt file that will be picked up by rails so it knows where to look for output files. 

My additional points above are two:

1) Before submitting a job, the rails code needs to find and remember the commit SHA for the pipeline branch, in order to ensure that the exact same pipeline code runs in all stages.

2) The pipeline version is directly available to the pipeline code and need not be passed through the environment.   In fact, passing it is more error-prone.